aiohttp/web_runner.py

```diff
@@ -238,6 +242,24 @@ def sites(self) -> Set[BaseSite]:
 
         self._server = await self._make_server()
 
+    async def serve_forever(self) -> None:
+        if self._serving_forever_fut is not None:
+            raise RuntimeError(
+                'Concurrent calls to serve_forever() are not allowed'
+            )
+        if self._server is None:
+            raise RuntimeError('Call setup() first')
+
+        loop = asyncio.get_event_loop()
+        self._serving_forever_fut = loop.create_future()
+        try:
+            await self._serving_forever_fut
+        except asyncio.CancelledError:
+            await self.cleanup()
```

Comment:
@asvetlov Am I right assuming that `runner.serve_forever()` should do `self.cleanup()` if being cancelled?

Reply:
The cleanup is questionable.
I think it deserves async context manager, e.g.
```
async with runner: 
    await runner.serve_forever()
``` 
Please remove.
Async context manager support requires a searate PR.